hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1623,13 +1624,51 @@ private void processSecondPasses(ArrayList<? extends SecondPass> secondPasses) {
 			return;
 		}
 
-		for ( SecondPass secondPass : secondPasses ) {
```

Comment:
Is there a reason to store these now-special SecondPasses in the generic SecondPass group?  If you notice one of the changes in 5.0 was to split these SecondPasses by type when they needed special handling or ordering.  Seems like this falls perfectly into that pattern.